test/xbgpu/test_engine.py

```diff
@@ -236,116 +271,157 @@ def _create_heaps(
+        return spead2.send.asyncio.InprocStream(spead2.ThreadPool(), queues, feng_stream_config)
 
     async def _send_data(
         self,
-        mock_recv_stream: spead2.InprocQueue,
-        recv_stream: spead2.recv.asyncio.Stream,
+        mock_recv_streams: list[spead2.InprocQueue],
+        mock_send_stream: list[spead2.InprocQueue],
+        corrprod_outputs: list[XOutput],
         *,
         heap_factory: Callable[[int], list[spead2.send.HeapReference]],
-        heap_accumulation_threshold: int,
         batch_indices: Iterable[int],
         timestamp_step: int,
         n_ants: int,
         n_channels_per_substream: int,
+        frequency: int,
         n_spectra_per_heap: int,
-    ) -> np.ndarray:
+    ) -> tuple[np.ndarray, list[int]]:
         """Send a contiguous stream of data to the engine and retrieve the results.
 
-        Each full accumulation requires `heap_accumulation_threshold` batches of
-        heaps. However, `batch_indices` is not required to contain full
-        accumulations.
+        Each full accumulation (for each corrprod-output) requires
+        `heap_accumulation_threshold` batches of heaps. However, `batch_indices`
+        is not required to contain full accumulations.
 
         Parameters
         ----------
         mock_recv_stream
             Fixture
-        recv_stream
-            InprocStream to receive data output by XBEngine.
+        mock_send_stream
+            Fixture
+        corrprod_outputs
+            Fixture
         heap_factory
             Callback that takes a batch index and returns the heaps for that index.
-        heap_accumulation_threshold
-            Number of consecutive heaps to process in a single accumulation.
         batch_indices
             Indices of the batches to send. These must be strictly increasing,
             but need not be contiguous.
         timestamp_step
             Timestamp step between each received heap processed.
-        n_ants, n_channels_per_substream, n_spectra_per_heap
+        n_ants, n_channels_per_substream, n_spectra_per_heap, frequency
             See :meth:`_create_heaps` for more info.
 
         Returns
         -------
         device_results
             Array of all GPU-generated data of shape
-            - (n_total_accumulations, n_channels_per_substream, n_baselines, COMPLEX)
+            - (len(corrprod_outputs), n_total_accumulations, n_channels_per_substream, n_baselines, COMPLEX)
+        n_accumulations_completed
+            Number of completed accumulations for each corrprod_output
         """
         max_packet_size = n_spectra_per_heap * N_POLS * COMPLEX * SAMPLE_BITWIDTH // 8 + PREAMBLE_SIZE
         max_heaps = n_ants * HEAPS_PER_FENGINE_PER_CHUNK * 10
-        feng_stream = self._make_feng(mock_recv_stream, max_packet_size, max_heaps)
+        feng_stream = self._make_feng(mock_recv_streams, max_packet_size, max_heaps)
+
+        acc_index_counts: list[dict[int, int]] = []
```

Comment:
Would be helpful to have a comment indicating what this structure represents.

Reply:
I think a comment here would still be helpful, even though it's now a list of sets.